train_gpt2.cu

```diff
@@ -2412,13 +2411,14 @@ float gpt2_update(GPT2 *model, float learning_rate, float beta1, float beta2, fl
     // AdamW update
     int block_size = 512;
     int num_blocks = CEIL_DIV(num_parameters, block_size);
-    float beta1_correction = 1.0f - powf(beta1, t);
-    float beta2_correction = 1.0f - powf(beta2, t);
     unsigned int seed = random_u32(&model->rng_state);
+    float learning_rate_inv_beta1_correction = learning_rate*(1.0f/1.0f - powf(beta1, t));
+    float inv_beta2_correction = 1.0f/(1.0f - powf(beta2, t));
+    float learning_rate_weight_decay = learning_rate * weight_decay;
+    assert(num_parameters % block_size == 0); //bounds check
```

Comment:
this check is cheap, essential for correctness, and whether it fails or not is dependent  on external program input, so maybe that should be a "real" check, and not an assert that may be turned off for a "release-mode" compile.